LBHackney-IT/lbh-social-care
# Added support for `isMulti` to the form components

**What**  
- simplified the `DynamicInput`
- introduced the `DynamicInputMulti` that is taking care of add multiple instances of the same input
- extracted the summary formatting login in an ad-hoc component `utils/summary` and added test coverage
- fixed `ObjestInput` test
- Reverted "removed summaryInline from the props"
- added phone number in `create-person` form
- small style tweaks

**How is working**
Every input now support `isMulti` and `isMultiTrigger`
```
{
  component: 'Object',
  name: 'phone_number',
  label: 'Phone Number',
  isMulti: true,
  components: [
    {
      component: 'PhoneInput',
      name: 'phoneNumber',
      label: 'Phone number',
      rules: { required: true },
    },
    {
      component: 'TextInput',
      name: 'phoneType',
      label: 'Phone type',
    },
  ],
},
```
is returning:
```
[
	  {
	    phoneType: "main contact",
	    phoneNumber: "077..."
	  },
      {
	    phoneType: "second contact",
	    phoneNumber: "077..."
	  },
	  ...
]
```

**Notes**
- At the moment is not possible to have a `isMulti` component in a `isMulti` step

**How to test it**
- check `create-person` form
or
- go to http://dev.hackney.gov.uk:3000/form/test/first-step
- select show object step
- check that is working properly

<img width="1004" alt="Screenshot 2021-01-27 at 18 36 12" src="https://user-images.githubusercontent.com/435399/106038087-c5efe300-60d7-11eb-89a1-9af51076b206.png">
<img width="1005" alt="Screenshot 2021-01-27 at 17 36 15" src="https://user-images.githubusercontent.com/435399/106038098-cb4d2d80-60d7-11eb-8152-7a2197850a88.png">
